x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -37,53 +41,113 @@ export const uniqueMonitorIds = (items: GetMonitorStatusResult[]): Set<string> =
     return acc;
   }, new Set<string>());
 
+const MESSAGE_AVAILABILITY_MAX = 3;
+
+export const availabilityMessage = (
+  availabilityResult: GetMonitorAvailabilityResult[],
+  threshold: string
+): string => {
+  return availabilityResult.length > 0
+    ? availabilityResult
+        .sort((a, b) => (a.availabilityRatio ?? 100) - (b.availabilityRatio ?? 100))
```

Comment:
I somehow missed this the first time through, but this map/reduce block is hard to read. I love map reduce, but a challenge chained large map reduce blocks brings is that with no intermediate variable names it's hard to understand semantically what each stage is doing.

There are a few ways to combat this

1. Add lots of comments
2. Assign the arguments to `map` and `reduce` to `const` functions that have good names
3. Use something other than map/reduce

Reply:
Attempted to improve this by moving the functions to named constants, and added some comments. https://github.com/elastic/kibana/pull/70284/commits/80c9983d469549c6087b28446d04e681bc2c26ac